### PR TITLE
Remove inconsequential if expression

### PR DESCRIPTION
**Pull request by RobertDober**
Hope this was not done intentionally :blush:

### CODE REVIEW

File: lib/nimble_parsec.ex [hunk 1]
```diff
@@ -158,14 +158,8 @@ defmodule NimbleParsec do
...
-        if combinator_kind == :def do
-          for {name, args, guards, body} <- defs do
-            def unquote(name)(unquote_splicing(args)) when unquote(guards), do: unquote(body)
-          end
-        else
-          for {name, args, guards, body} <- defs do
-            defp unquote(name)(unquote_splicing(args)) when unquote(guards), do: unquote(body)
-          end
+        for {name, args, guards, body} <- defs do
+          def unquote(name)(unquote_splicing(args)) when unquote(guards), do: unquote(body)
```
Comment (josevalim) [inline lib/nimble_parsec.ex:162]: It is a small detail but one is `def`, the other is `defp`. :)

Comment (RobertDober) [reply lib/nimble_parsec.ex:162]: @josevalim  I guess I shall apologize - need new glasses - but actually the tests still passed, shall I change the PR to just add a test for that case (and leave the code unchanged of course)?

Comment (RobertDober) [reply lib/nimble_parsec.ex:162]: @josevalim  maybe I am wrong, but I guess there is something wrong here (to not abuse of the proverbial smell in a scandinavian country as mentioned in a play by Shakespear)

I just put an `IO.inspect` at the end of the `for`:

```
[parse__0: 6, parse__0: 6, parse__1: 6]
[parsep__0: 6, parsep__0: 6, parsep__1: 6]
[combinator__0: 6, combinator__0: 6, combinator__1: 6]
[combinatorp__0: 6, combinatorp__0: 6, combinatorp__1: 6
```
 is the AST that gets created here

While the b

Comment (josevalim) [reply lib/nimble_parsec.ex:162]: A PR that improves the coverage would certaily be welcome!

Comment (RobertDober) [reply lib/nimble_parsec.ex:162]: I will try...

**Pull request closed without merging**